### DISCUSSION TITLE
Tabbed interface

### DISCUSSION THREAD

**Discussion by LuisVilaplana** [oqtane/oqtane.framework#3300] [General] [answered]
Does anyone have any idea how to create a theme that simulates a tabbed interface for Oqtane? What I'm looking for is that every time an option is selected in the menu, this module opens within its own tab on the same page, so that the user can have multiple modules active at the same time for better productivity.

I have managed to do this in Blazor, but using a Tab component from a library called Bootstrap Blazor.

![image](https://github.com/oqtane/oqtane.framework/assets/103559699/427833f5-59d5-4bdf-a9fd-d01faee9c8b6)

Thanks

**Comment by sbwalker** [comment]
Is there a reason this has to be done in a theme? A theme is simply a container - so it does not need to have any content other than a Pane component. You could implement this in a module using a TabStrip and TabPanel component - there are plenty of examples in the Admin UI which uses this approach. If you need to implement it in the theme you could also simply use Bootstrap classes directly (as this is all the TabStrip and TabPanel use internally) - https://getbootstrap.com/docs/5.0/components/navs-tabs/#tabs

**Comment by LuisVilaplana** [comment]
I can't see a way to have different pages open in the same page using a module with Tabstrip. When you select them from the menu, the current page is removed, and a new one opens. That's why I was thinking of doing it in a theme because the theme persists across different pages.

In any case, is there a way to open a page in another tab of the same browser in Oqtane without right-clicking and selecting "Open in new tab"?

**Comment by sbwalker** [comment, marked as answer]
I did not realize you wanted to open different Oqtane pages based on clicking different tabs. This could be accomplished by building a custom Menu component which displays tabs instead of links (like the current horizontal menu) and using it in your custom theme. Or you could just embed the tab rendering logic directly into your custom theme. Either way this is not difficult.

**Reply by LuisVilaplana** [reply]
Okay, I'm going to try that, thank you very much. For desktop users, I plan to use PWA, and now I can also use the new multi-tabs feature. This opens a new tab within the same Webview, directly to the main page of the site, which can be a menu of options. However, for browsers or other devices, I'll have to find another method. I believe with a custom menu that opens pages in another tab, I can solve it. 

![image](https://github.com/oqtane/oqtane.framework/assets/103559699/7fdaf3df-de1b-4e37-aa2c-bb4e73ba168f)

Thanks again
